en/docs/api-analytics/samples/publishing-custom-analytics-data.md

```diff
@@ -0,0 +1,193 @@
+        ```
+        3. Point local m2 repository in project pom
+        ```code
+        <repository>
+            <id>local-maven-repo</id>
+            <url>file://home/user/.m2/repository</url>
+        </repository>
+        ```
+        Follow the URL pattern when providing the repo url
+
+### Implementing Required Class
+
+#### AnalyticsCustomDataProvider Implementation Class
+
+Custom analytics data can be added to the existing event schema with the use of a class of type `AnalyticsCustomDataProvider`. Therefore, it is required to implement a class from the `AnalyticsCustomDataProvider` interface. By overriding the `getCustomProperties` method it is possible to push custom analytics data.
+
+In order to achieve this behavior, create a class implementing the `AnalyticsCustomDataProvider` Interface of `org.wso2.carbon.apimgt.common.analytics.collectors`.
+
+Implementation of this class should look something similar to [this](https://github.com/wso2/samples-apim/blob/4.0.0/analytics-custom-data-provider/src/main/java/org/wso2/carbon/apimgt/gateway/sample/publisher/CustomDataProvider.java).
+
+#### Build the Project
+
+Build the project using,
+
+    mvn clean install
+
+## Configuring the Sample
+
+This section will cover the steps required to configure WSO2 API-M Gateway for the sample created above. The steps covered are adding the .jar file, configuring the deployment.toml file, and enabling the logs.
+
+!!! Warning
+    Note that WSO2 API Manager 3.0.0, 3.1.0, 3.2.0, and 4.0.0 are affected by the **Log4j2 zero-day** vulnerability, which has been reported to WSO2 on 10th December 2021. You can mitigate this vulnerability in your product by following our [instructions and guidelines](https://docs.wso2.com/pages/viewpage.action?pageId=180948677).
+
+1. Adding the .jar file.
+
+    Place the created .jar file inside the `wso2am-4.0.0/repository/components/lib` directory.
+
+2. Configuring the deployment.toml file.
+
+    Edit `apim.analytics` configurations in the `deployment.toml` file located inside `wso2am-4.0.0/repository/conf` with the following configuration.
+
+        [apim.analytics]
+        enable = true
+        properties."publisher.custom.data.provider.class" = "<FullyQualifiedClassNameOfAnalyticsCustomDataProviderImplClass>"
+        type = "elk"
+
+    !!! Important
+        Type should be given as `elk` as this property value is filtered out in cloud implementation.
+
+3. Enabling Logs
+
+    To [enable trace logs]({{base_path}}/administer/logging-and-monitoring/logging/configuring-logging/#enabling-logs-for-a-component) for the component: `org.wso2.am.analytics.publisher`, edit `log4j2.properties` file located inside `wso2am-4.0.0/repository/conf` directory. 
+
+    a) Add a reporter to the loggers list:
```

Comment:
Shall we use standard markdown numbering for this? Add it as 1. and 2. and it will render as sub points.

Reply:
Fixed with https://github.com/wso2/docs-apim/pull/6104/commits/1a243d5c12d13171c6a00eecdba75d374c2bc77c
Thanks